src/gui/toolbarMenubar/model/ToolbarEntry.cpp

```diff
@@ -1,18 +1,27 @@
 #include "ToolbarEntry.h"
 
+#include <algorithm>
 #include <utility>
 
 ToolbarEntry::ToolbarEntry() = default;
 
 ToolbarEntry::ToolbarEntry(const ToolbarEntry& e) { *this = e; }
+ToolbarEntry::ToolbarEntry(ToolbarEntry&& e) { *this = e; }
 
-void ToolbarEntry::operator=(const ToolbarEntry& e) {
+ToolbarEntry& ToolbarEntry::operator=(const ToolbarEntry& e) {
     this->name = e.name;
     clearList();
 
     for (ToolbarItem* item: e.entries) {
         entries.push_back(new ToolbarItem(*item));
     }
+    return *this;
+}
+
+ToolbarEntry& ToolbarEntry::operator=(ToolbarEntry&& e) {
+    this->name = std::move(e.name);
+    this->entries = std::move(e.entries);
+    return *this;
 }
```

Comment:
Yes when you accidently move an reference of the object irrelevant from which place it came, on itself, you will clear the object and then you'll move an cleared object. This results in bugs which are extremely hard to find.

For reference google c++ self assignment.

Reply:
Well, that's not really a problem with the code but more like a problem with C++.

Reply:
The thing is, that for both the copy and the move assignment operator override an existing Object, this means, that all special state must be cleared before override. In our case we must call `clearLists` before (move)assigning.
But this can result into bugs, when we use the same object as source and target:

```
Object k;
some_code_passing_k_als_reference(k);
some_other_code();


auto new_o = get_an_objectref_from_other_loc(); // This is surprisingly k
k = new_o;
```

After the assignment k will be an invalid wrong object.

Reply:
I see the problem now. I have fixed the error.